crates/router/src/connector/klarna/transformers.rs

```diff
@@ -73,13 +83,40 @@ pub struct KlarnaPaymentsRequest {
     shipping_address: Option<KlarnaShippingAddress>,
 }
 
+#[derive(Default, Debug, Serialize, Deserialize)]
+pub struct CheckoutRequest {
+    auto_capture: bool,
+    order_lines: Vec<CheckoutOrderLines>,
+    order_amount: MinorUnit,
+    purchase_country: enums::CountryAlpha2,
+    purchase_currency: enums::Currency,
+    shipping_address: Option<KlarnaShippingAddress>,
+    order_tax_amount: MinorUnit,
+    merchant_urls: MerchantURLs,
+}
+
 #[derive(Debug, Deserialize, Serialize)]
-pub struct KlarnaPaymentsResponse {
+#[serde(untagged)]
+pub enum KlarnaAuthResponse {
```

Comment:
If possible, try not to use untagged responses

Reply:
Not using untagged will cause deserialize error, as only required fields are mentioned inside the struct, even in case of **klarnaSDK** as well. So using untagged is more appropriate as of now.